subprojects/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/tasks/GeneratePrecompiledScriptPluginAccessors.kt

```diff
@@ -487,3 +521,43 @@ fun ProjectInternal.applyPlugins(pluginRequests: PluginRequests) {
         classLoaderScope
     )
 }
+
+
+private
+fun <T> Gradle.withLogLevel(logLevel: LogLevel, block: () -> T): T {
+    val loggingManager = serviceOf<LoggingManagerInternal>()
+    val previousLevel = loggingManager.level
+    return try {
+        loggingManager.setLevelInternal(logLevel)
+        block()
+    } finally {
+        loggingManager.setLevelInternal(previousLevel)
+    }
+}
+
+
+private
+fun <T> withCapturedOutputOnError(block: () -> T, onError: (ErrorWithCapturedOutput) -> T): T {
+    val outCapture = ByteArrayOutputStream()
+    val errCapture = ByteArrayOutputStream()
+    return try {
+        val previousOut = System.out
+        val previousErr = System.err
+        try {
+            System.setOut(PrintStream(outCapture))
```

Comment:
This is not thread-local behavior, you are redirecting the streams for the whole JVM, right? Isn't that an issue? Or is it still better than the alternative?